src/main/java/httpserver/server/SocketHandler.java

```diff
@@ -15,10 +16,12 @@
     Socket clientSocket;
     IRequestParser requestParser;
     IResponseWriter responseWriter;
+    Router router;
 
-    public SocketHandler(IRequestParser requestParser, IResponseWriter responseWriter){
+    public SocketHandler(IRequestParser requestParser, IResponseWriter responseWriter, Router router){
         this.requestParser = requestParser;
         this.responseWriter = responseWriter;
+        this.router = router;
     }
```

Comment:
Great job adding router as a collaborator! Do you see how you have interfaces on the first two collaborators but a concretion for Router? Which SOLID principle is violated here? How can you fix it?